Homework13/Tests/WebTableTests.cs

```diff
@@ -0,0 +1,139 @@
+            Assert.Multiple(() =>
+            {
+                Assert.AreEqual(filteredResultRowsCountWithSearchedValue, resultRowsCountWithSearchedValue);
+                Assert.AreEqual(filteredRowsCountWithSearchedValue, resultRowsCountWithSearchedValue);
+            });
+        }
+
+        // Test3: Add new entry.
+        [Test]
+        public void AddNewEntry()
+        {
+            GenericPages.WebTablePage.ClickAddNewEntryButton();
+
+            // Calculate estimated new row id.
+            var numberOfExistingRows = GenericPages.WebTablePage.GetListOfNotEmptyRows().Count;
+            var indexOfNewEntryRow = numberOfExistingRows + 2;
+
+            // Fill in form inputs.
+            GenericPages.WebTablePage.FillInFirstNameTextBox(_firstName);
+            GenericPages.WebTablePage.FillInLastNameTextBox(_lastName);
+            GenericPages.WebTablePage.FillInEmailTextBox(_email);
+            GenericPages.WebTablePage.FillInAgeTextBox(_age);
+            GenericPages.WebTablePage.FillInSalaryTextBox(_salary);
+            GenericPages.WebTablePage.FillInDepartmentTextBox(_department);
+            GenericPages.WebTablePage.ClickSubmitButton();
+
+            // Get text of new entry and compare it with expected.
+            var entryValue = $"{_firstName} {_lastName} {_age} {_email} {_salary} {_department}";
+            Assert.AreEqual(entryValue, GenericPages.WebTablePage.GetTextOfElement(indexOfNewEntryRow));
+        }
+
+        // Test4: Edit first entry.
+        [Test]
+        public void EditFirstEntry()
+        {
+            // Open edit form for the first entry.
+            var firstName1 = GenericPages.WebTablePage.GetFirstName1Value();
+            GenericPages.WebTablePage.ClickEditEntryButton(firstName1);
+
+            // Clear form inputs.
+            GenericPages.WebTablePage.ClearFirstNameTextBox();
+            GenericPages.WebTablePage.ClearLastNameTextBox();
+            GenericPages.WebTablePage.ClearEmailTextBox();
+            GenericPages.WebTablePage.ClearAgeTextBox();
+            GenericPages.WebTablePage.ClearSalaryTextBox();
+            GenericPages.WebTablePage.ClearDepartmentTextBox();
+
+            // Fill in form inputs.
+            GenericPages.WebTablePage.FillInFirstNameTextBox(_firstName);
+            GenericPages.WebTablePage.FillInLastNameTextBox(_lastName);
+            GenericPages.WebTablePage.FillInEmailTextBox(_email);
+            GenericPages.WebTablePage.FillInAgeTextBox(_age);
+            GenericPages.WebTablePage.FillInSalaryTextBox(_salary);
+            GenericPages.WebTablePage.FillInDepartmentTextBox(_department);
+            GenericPages.WebTablePage.ClickSubmitButton();
+
+            // Get text of updated entry and compare it with expected.
+            var entryValue = $"{_firstName} {_lastName} {_age} {_email} {_salary} {_department}";
+            Assert.AreEqual(entryValue, GenericPages.WebTablePage.GetTextOfElement(2));
```

Comment:
It's not clear from test why we pass 2 if we want to get values of the first row